<launch>
    <arg name="demands_topic" default="listen" />
    <arg name="say_topic" default="say" />
    <arg name="command_topic" default="command" />
    <arg name="response_topic" default="response" />
    <arg name="translation_topic" default="translation" />
    

    <!-- Launch Hardware -->
    <!-- <include file="$(find spot_driver)/launch/driver.launch"/> -->

    <!-- Launch Mapping & Navigation-->
    <!-- <include file="$(find nurse)/launch/nav.launch"/> -->
       
    <!-- Launch Object Detection -->

    <!-- Launch Audio Interpretation -->
    <include file="$(find nurse)/launch/audio.launch" >
        <arg name="demands_topic" value="$(arg demands_topic)"/>  
        <arg name="say_topic" value="$(arg say_topic)" />
        <arg name="command_topic" value="$(arg command_topic)" />
        <arg name="response_topic" value="$(arg response_topic)" />
        <arg name="translation_topic" value="$(arg translation_topic)" />
    </include>

    <!-- Launch Voice for Spot -->
    <include file="$(find nurse)/launch/voice.launch" >
        <arg name="demands_topic" value="$(arg demands_topic)"/>  
        <arg name="say_topic" value="$(arg say_topic)" />
        <arg name="path_to_media" value="../workspace/aiil_workspace/noetic_workspace/src/spot/nurse/media" />
    </include>

    <!-- Launch Brain node for Spot -->
    <include file="$(find nurse)/launch/brain.launch" >
        <arg name="say_topic" value="$(arg say_topic)" />
        <arg name="command_topic" value="$(arg command_topic)" />
        <arg name="response_topic" value="$(arg response_topic)" />
    </include>    
</launch>
